MM-316 Allows users to drag and drop files from their computer to the center pane or RHS to upload them

rgarmsen2295 opened the pull request:
Also fixes an issue with uploading files to the RHS after switching channels along with some minor cosmetic reformatting of file_upload.jsx.

I added a small jquery 'plugin' to help handle the actual drag and drop detection rather than use the not so great default html5 implementation (it's really not good).

Finally, thanks to @asaadmahmoodspin for the UI changes!

ralder commented on web/sass-files/sass/partials/_post.scss:
```diff
@@ -106,6 +106,39 @@ body.ios {
+	}
+
+	.invisible {
+		visibility: hidden;
```
this rule already exists in bootstrap

ralder commented on web/sass-files/sass/partials/_post.scss:
```diff
@@ -106,6 +106,39 @@ body.ios {
+		visibility: hidden;
+	}
+	.visible {
+		visibility: visible;
```
seems this rule not need because may use just toggle classname `.invisible`

rgarmsen2295 replied:
Thanks for the comments @ralder I think you're right for both of these let me take a look and see about changing these.

rgarmsen2295 commented:
Okay thanks again @ralder, implemented your suggestions and fixed an issue that popped up with @asaadmahmoodspin's help.  @jwilander please review at your leisure, thanks!

jwilander commented:
Code looks good to me. What browsers was this tested on?

rgarmsen2295 commented:
Chrome on osx, safari, and IE10 through a mac vm running win7; part of the reason I went with the jquery version of the plugin was the better cross-browser support.

jwilander commented:
Perfect. Just waiting for a seconder

crspeller commented:
Looks good!
